crypto/rand/drbg_lib.c

```diff
@@ -177,12 +177,17 @@ int RAND_DRBG_set(RAND_DRBG *drbg, int type, unsigned int flags)
         else
             ret = drbg_hash_init(drbg);
     } else {
+        drbg->type = 0;
+        drbg->flags = 0;
+        drbg->meth = NULL;
         RANDerr(RAND_F_RAND_DRBG_SET, RAND_R_UNSUPPORTED_DRBG_TYPE);
         return 0;
     }
 
-    if (ret == 0)
+    if (ret == 0) {
+        drbg->state = DRBG_ERROR;
```

Comment:
+1: That's a good idea to explicitly set the error state. Since `RAND_DRBG_set()` is called by
`RAND_DRBG_uninstantiate()`, you should probably also set the error state there for consistency reasons:

https://github.com/openssl/openssl/blob/97798f460a785253fea01a9ab02fc2100e7f85d5/crypto/rand/drbg_lib.c#L438-L442